### PR TITLE
Add centralized documentation for global rlang options.

### LINKED ISSUE #899: Centralized documentation for options that affect rlang behaviour
Is there a central place to find all of rlang's options and what they do?

This is motivated by my recent journey into options that affect back traces (re: knitr, reprex, etc.).

If not, can there be? And would it be a good issue for Tidyverse dev day?

I made a [help topic like this for gargle](https://gargle.r-lib.org/reference/gargle_options.html). I realize options might play a different role here in rlang, but maybe the basic idea is still sound.

### PR DESCRIPTION
**Pull request by smingerson**
Added a centralized doc location for global options, accessible through `?rlang::rlang_options`. It is pretty brief, and points to more detailed documentation, if it exists. 

Fixes #899

### CODE REVIEW

File: R/rlang-options.R [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#' Global options for rlang
+#'
+#' @description
+#' rlang has several options which may be set globally to control
+#'  behavior. A brief description of each is given here. If any functions
```
Comment (lionel-) [inline R/rlang-options.R]: Can you remove the extra leading space please?

File: R/rlang-options.R [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+#'  are referenced, refer to their documentation for additional details.
+#'
+#' * rlang_interactive: A logical value used by [is_interactive()]. This
```
Comment (lionel-) [inline R/rlang-options.R]: The option should be backticked so they appear like code.

File: R/rlang-options.R [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+#'    can be set to `TRUE` to test interactive behavior in unit tests,
+#'    for example.
+#' * rlang_backtrace_on_error: A character string which controls whether
```
Comment (lionel-) [inline R/rlang-options.R]: Needs newlines between bullet items to improve readability.

File: R/rlang-options.R [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+#'    backtraces are displayed with error messages, and the level of
+#'    detail they print. See [rlang_backtrace_on_error] for the possible option values.
+#' * rlang_trace_format_srcrefs: A logical value used to control whether
+#'    srcrefs are printed as part of the backtrace.
+#' * rlang_trace_top_env: An environment which will be treated as the
+#'    top-level environment when printing traces. See [trace_back()]
+#'    for examples.
+#' @name rlang_options
```
Comment (lionel-) [inline R/rlang-options.R]: I think the topic could be called `faq-options`, this will ensure it's included in the pkgdown reference along other faq items.

Comment (smingerson) [reply R/rlang-options.R]: I've left the file as `rlang-options.R`, but am happy to change that as well.

Comment (lionel-) [reply R/rlang-options.R]: Thanks, `faq.R` might be appropriate.

Comment (smingerson) [reply R/rlang-options.R]: The rlang pkgdown does not have a FAQ section, though it does have  "session state", which includes `matches("option")`. I had thought that had been the section you were referring to. 

Would you like me to update the pkgdown so it does have a FAQ section, similar to what [tidyselect](https://tidyselect.r-lib.org/reference/index.html#section-faq) has?

Comment (lionel-) [reply R/rlang-options.R]: That'd be great, thanks!

File: _pkgdown.yml [hunk 5]
```diff
@@ -1,277 +1,281 @@
...
+      - new_data_mask
+      - as_data_pronoun
+  - title: Condition handlers
+    contents:
+      - with_handlers
+      - calling
+  - title: Session state
+    contents:
+      - matches("interactive")
+      - matches("^[^(faq)]option")
```
Comment (smingerson) [inline _pkgdown.yml]: That regex doesn't actually do what I want. I think the `matches("option")` should be changed to `ends_with("option")` to pick up all of the options documentation without faq-options. Unless we want the global options to be documented under both session state and FAQ

Comment (lionel-) [reply _pkgdown.yml]: Yes it seems fine to document in both sections.

Can you double check your changes to the pkgdown file please? It seems you have reindented everything. The commit should not contain any whitespace changes.

Comment (smingerson) [reply _pkgdown.yml]: All fixed up, sorry about that.

**Comment by lionel-** [comment]
That's great, thanks!

**Pull request merged**